### PR TITLE
Only regenerate certificate fixtures if they do not exist

### PR DESCRIPTION
**Pull request by ehelms**
This allows adding more certificate fixtures without having to
regenerate all certificates every time and leading to larger
diffs.

After doing a set of certificate testing PRs, I would like to get this re-factor in first to reduce the diffs.

### CODE REVIEW

File: spec/fixtures/katello-certs-check/certs/foreman.example.com.csr [hunk 1]
```diff
@@ -0,0 +1,15 @@
+-----BEGIN CERTIFICATE REQUEST-----
```
Comment (ekohl) [inline spec/fixtures/katello-certs-check/certs/foreman.example.com.csr]: Can we keep this out? Perhaps the `.gitignore` rules need to be moved/updated?

File: spec/fixtures/katello-certs-check/certs/ca.srl [hunk 2]
```diff
@@ -0,0 +1 @@
+7AD3BECEFC09E272CDAF17BD95CAD3E21D27A023
```
Comment (ekohl) [inline spec/fixtures/katello-certs-check/certs/ca.srl]: Is the serial needed to be stored in git?

File: spec/katello_certs_check_spec.rb [hunk 3]
```diff
@@ -5,35 +5,39 @@
...
-  it 'without parameters' do
-    stdout, stderr, status = Open3.capture3(command)
-    expect(stderr).to eq fixture('missing-parameter.txt')
-    expect(stdout).to eq ''
-    expect(status.exitstatus).to eq 1
-  end
+  let(:command) { File.join(__dir__, '..', 'bin', 'katello-certs-check') }
+  let(:directory) { File.join(FIXTURE_DIR, 'katello-certs-check') }
+  let(:certs_directory) { File.join(directory, 'certs') }
+  let(:ca) { File.join(certs_directory, 'ca.crt') }
```
Comment (ekohl) [inline spec/katello_certs_check_spec.rb:15]: Is it needed to move these from where they were? Moving things around makes a diff harder to read and thus more work for a reviewer.

**Review by ekohl — CHANGES_REQUESTED** [review]

File: spec/katello_certs_check_spec.rb (hunk 3 above)
Comment (ehelms) [reply spec/katello_certs_check_spec.rb:15]: Do you mean the certificates themselves or do you mean these lines in the test? In both cases,  For the former, yes. For the latter, strictly needed? No. Does it help to draw a clearer line for the context? Yes. I am also looking ahead at PRs that I have that will all need to make this structural change. And it being easier to do so here, and provide additional context with the change.

Comment (ehelms) [reply spec/katello_certs_check_spec.rb:15]: I dunno if it helps, but Github provides a method to remove whitespace changes from the diff when reviewing, e.g.

https://github.com/theforeman/foreman-installer/pull/618/files?diff=unified&w=1

**Review by ekohl — APPROVED** [review]

**Pull request merged**